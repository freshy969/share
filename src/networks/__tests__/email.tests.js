import Email from '../email';

describe('Share > Email', () => {
  test('url', () => {
    const email = Email.create({});
    Object.defineProperty(email, 'entityLink', {value: jest.fn(), writable: true});
    email.entityLink.mockReturnValueOnce('https://demo.frontity.com/the-beauties-of-gullfoss');

    expect(
      email.url({
        type: 'post',
        id: 60,
        subject: 'The Beauties of Gullfoss',
        body:
          'Gullfoss is a waterfall located in the canyon of the Hvítá river in southwest Iceland.',
      }),
    ).toMatchSnapshot();
  });
});
