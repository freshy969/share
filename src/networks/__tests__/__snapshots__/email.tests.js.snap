// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Share > Email url 1`] = `"mailto:?to=&subject=The%20Beauties%20of%20Gullfoss&body=Gullfoss%20is%20a%20waterfall%20located%20in%20the%20canyon%20of%20the%20Hv%C3%ADt%C3%A1%20river%20in%20southwest%20Iceland.%0A%0Ahttps%3A%2F%2Fdemo.frontity.com%2Fthe-beauties-of-gullfoss"`;
